### PR TITLE
Feature/hatoba firewall group

### PR DESCRIPTION
**Pull request by aokumasan**
## Description

- Add nifcloud_hatoba_firewall_group resource.

## How Has This Been Tested?

- [x]  Buld with `make install` command, and Apply example.

```:console
make install
cd examples/hatoba_firewall_group
terraform init -plugin-dir ~/.terraform.d/plugins
terraform plan
terraform apply
```

- [x] Run acceptance tests.

```:console
TF_ACC=1 go test ./nifcloud/acc/... -v -count 1 -parallel 20 -timeout 360m -run TestAcc_HatobaFirewallGroup
```

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

### CODE REVIEW

File: nifcloud/acc/hatoba_firewall_group_test.go [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+		return nil
+	}
+}
+
+func testAccHatobaFirewallGroupResourceDestroy(s *terraform.State) error {
+	svc := testAccProvider.Meta().(*client.Client).Hatoba
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "nifcloud_hatoba_security_group" {
```
Comment (miyuush) [inline nifcloud/acc/hatoba_firewall_group_test.go:240]:
```suggestion
		if rs.Type != "nifcloud_hatoba_firewall_group" {
```

File: docs/resources/hatoba_firewall_group.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+* `rule` - (Optional) A list of the Hatoba firewall group rule objects. see [rule](#rule).
+
+### rule
+
+#### Arguments
+
+* `cidr_ip` - (Required) The CIDR IP address that allow access.
+* `description` - (Optional) The firewall group rule description.
+* `direction` - (Optional) The direction of rule being created. Valid options are IN (Incoming) or OUT (Outgoing).
+* `from_port` - (Optional) The start port.
```
Comment (miyuush) [inline docs/resources/hatoba_firewall_group.md:58]:
```suggestion
* `from_port` - (Required) The start port.
```

File: docs/resources/hatoba_firewall_group.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+* `protocol` - (Optional) The protocol.
+* `to_port` - (Required if protocol is undefined) The end port.
```
Comment (miyuush) [inline docs/resources/hatoba_firewall_group.md:60]:
```suggestion
* `to_port` - (Optional) The end port.
```

File: nifcloud/acc/hatoba_firewall_group_test.go (hunk 1 above)
Comment (aokumasan) [reply nifcloud/acc/hatoba_firewall_group_test.go]: thanks! applied!!

File: docs/resources/hatoba_firewall_group.md (hunk 3 above)
Comment (aokumasan) [reply docs/resources/hatoba_firewall_group.md:60]: Thanks, this is my mistake 😇

File: docs/resources/hatoba_firewall_group.md (hunk 2 above)
Comment (aokumasan) [reply docs/resources/hatoba_firewall_group.md:58]: Parameters other than `cidr_ip` are modified to be optional because they have complex conditions... 😇

Comment (miyuush) [reply docs/resources/hatoba_firewall_group.md:58]: That's better.

**Review by miyuush — APPROVED** [review]
Thanks for the fix.
LGTM!

**Pull request merged**